src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/ProjectDesignerService.cs

```diff
@@ -16,19 +16,18 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
     internal class ProjectDesignerService : IProjectDesignerService
     {
         private readonly IUnconfiguredProjectVsServices _projectVsServices;
+        private readonly bool _supportsProjectDesigner;
 
         [ImportingConstructor]
         public ProjectDesignerService(IUnconfiguredProjectVsServices projectVsServices)
         {
             Requires.NotNull(projectVsServices, nameof(projectVsServices));
 
             _projectVsServices = projectVsServices;
+            _supportsProjectDesigner = _projectVsServices.VsHierarchy.GetProperty(VsHierarchyPropID.SupportsProjectDesigner, defaultValue: false);
```

Comment:
From my understanding, this value is computed through [this](http://index/?rightProject=Microsoft.VisualStudio.ProjectSystem.VS.Implementation&file=PropertyPages%5cVsProjectDesignerPageService.cs&line=50) property, which basically checks if there are any ProjectDesignerPageProviders exported at UnconfiguredProject level, which cannot change during the lifetime of an unconfigured project.